clang/lib/AST/PreorderAST.cpp

```diff
@@ -230,6 +274,52 @@ void PreorderAST::ConstantFold(Node *N, bool &Changed) {
   Changed = true;
 }
 
+void PreorderAST::NormalizeExprsWithoutConst(Node *N) {
+  // Normalize expressions by adding an integer constant to expressions which
+  // do not have one. For example:
+  // p ==> (p + 0)
```

Comment:
How does `p` become normalized to `p + 0`?

Reply:
I have added more comments inline explaining why we need to normalize `p` to `p + 0`.